salt/modules/aptpkg.py

```diff
@@ -3006,7 +3014,7 @@ def file_dict(*packages, **kwargs):
     return __salt__["lowpkg.file_dict"](*packages)
 
 
-def expand_repo_def(**kwargs):
+def _expand_repo_def(os_name, lsb_distrib_codename=None, **kwargs):
```

Comment:
We are changing behavior to this function by now requiring `os_name` and `lsb_distrib_codename`. If someone were calling `expand_repo_def` previously then they would have to update their kwargs/args without warning.

Reply:
This function is now private.
I can keep the old one, which has the deprecation, with the precious signature, since it has access to dunders....

Reply:
that works

Reply:
In the function that is getting deprecated can we set os_name and lsb_distrib_codename from `__grains__` if its not in kwargs?

Reply:
https://github.com/saltstack/salt/pull/62837